integration/query_comparision_test.go

```diff
@@ -170,6 +170,29 @@ func TestQueryComparisonEq(t *testing.T) {
 			q:           bson.D{{"value", bson.D{{"$eq", int64(math.MinInt64)}}}},
 			expectedIDs: []any{"int64-min"},
 		},
+		"CompareNoSuchField": {
+			q: bson.D{{"no-such-field", nil}},
```

Comment:
`{"no-such-field": {"$eq": nil}}` is missing.

And please rename test cases and group them by blank lines just like other cases